packages/core/src/server/auth/sessions.ts

```diff
@@ -655,6 +656,9 @@ export async function getSessionKernel(
     if (enableCsrfProtection && payload.antiCSRFToken !== antiCSRFToken) {
       // await revokeSession(req, res, payload.handle, true)
       setHeader(res, HEADER_CSRF_ERROR, "true")
+      log.warning(
+        "Request must be performed using an anti-csrf token, you can learn more here : https://blitzjs.com/docs/session-management#manual-api-requests",
```

Comment:
Awesome, thank you so much!!

For this actual warning, let's move it to the top and only log it if the header is actually missing.

So something like this:

```tsx
  const antiCSRFToken = req.headers[HEADER_CSRF] as string

  if (enableCsrfProtection && !antiCSRFToken) {
    log.warning(`This request is missing the ${HEADER_CSRF} header. You can learn about adding this here: https://blitzjs.com/docs/session-management#manual-api-requests`)
  }
```

Reply:
👍 I like that this has a link to the docs rather than trying to dump all of that information into your logs